lab_1/main.py

```diff
@@ -86,9 +134,28 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
+    if isinstance(unknown_profile, dict) and isinstance(profile_to_compare, dict) and isinstance(top_n, int):
```

Comment:
There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.